19.3/basic/app.py

```diff
@@ -0,0 +1,72 @@
+from flask import Flask, request, render_template, redirect, flash, session
+from flask_debugtoolbar import DebugToolbarExtension
+from surveys import satisfaction_survey as survey
+
+RESPONSES_KEY = "responses"
+
+app = Flask(__name__)
+app.config['SECRET_KEY'] = "never-tell!"
+app.config['DEBUG_TB_INTERCEPT_REDIRECTS'] = False
+
+debug = DebugToolbarExtension(app)
+
+@app.route("/")
+def start_page():
+    """Home page to select a survey"""
+    
+    return render_template("start-page.html", survey=survey)
+
+@app.route("/begin", methods=["POST"])
+def start_survey():
+    """Clear session responses"""
+    session[RESPONSES_KEY] = []
+    return redirect("/questions/0")
+
+@app.route("/answer", methods=["POST"])
+def handle_question():
+    """Save response and redirect to next question."""
+
+    # get the response choice
+    choice = request.form['answer']
+
+    # add response to session instead of appending to list
+    responses = session[RESPONSES_KEY]
+    responses.append(choice)
+    session[RESPONSES_KEY] = responses
+
+    if (len(responses) == len(survey.questions)):
+        # They've answered all the questions! Thank them.
+        return redirect("/complete")
+
+    else:
+        return redirect(f"/questions/{len(responses)}")
+
+
+@app.route("/questions/<int:qid>")
```

Comment:
@Danielewin8 also, see how they are passing the qid here, you could use something similar to this in Boggle exercise for word.